R/bootstrap.R

```diff
@@ -382,11 +374,78 @@ renv_bootstrap_download_github <- function(version) {
     return(FALSE)
   }
 
+  renv_bootstrap_download_augment(destfile)
+
   catf("OK")
   return(destfile)
 
 }
 
+# Add Sha to DESCRIPTION. This is stop gap until #890, after which we
+# can use renv::install() to fully capture metadata.
+renv_bootstrap_download_augment <- function(destfile) {
+  sha <- renv_bootstrap_git_extract_sha1_tar(destfile)
+  if (is.null(sha)) {
+    return()
+  }
+
+  # Untar
+  tempdir <- tempfile("renv-github-")
+  on.exit(unlink(tempdir, recursive = TRUE), add = TRUE)
+  untar(destfile, exdir = tempdir)
+  pkgdir <- dir(tempdir, full.names = TRUE)[[1]]
+
+  # Modify description
+  desc_path <- file.path(pkgdir, "DESCRIPTION")
+  desc_lines <- readLines(desc_path)
```

Comment:
We should remove potential trailing newlines from the DESCRIPTION here, lest we incur the wrath of `Error: contains a blank line`

Reply:
I can't quite picture how we'd get a blank line here if there wasn't one in the original `DESCRIPTION`. Do you have a specific scenario that you're picturing?

Reply:
My worry here is that if the DESCRIPTION had trailing newlines, and then we appended to that content, we'd end up writing a DESCRIPTION file with newlines between fields. R allows the blank trailing newlines in a DESCRIPTION file (and in my experience this is kind of common), but not the blank lines between fields.

Reply:
```
> writeLines(c(readLines("DESCRIPTION"), "RemoteType: github"))
Package: renv
Type: Package
Title: Project Environments
Version: 0.17.3-62
Authors@R: c(
    person("Kevin", "Ushey", role = c("aut", "cre"), email = "kevin@rstudio.com"),
    person("Posit Software, PBC", role = c("cph", "fnd"))
    )
    
Description: A dependency management toolkit for R. Using 'renv', you can create
    and manage project-local R libraries, save the state of these libraries to
    a 'lockfile', and later restore your library as required. Together, these
    tools can help make your projects more isolated, portable, and reproducible.
License: MIT + file LICENSE
URL: https://rstudio.github.io/renv/
BugReports: https://github.com/rstudio/renv/issues
Imports: utils
Suggests: BiocManager, cli, covr, cpp11, devtools, gitcreds, jsonlite, knitr, miniUI,
    packrat, pak, R6, remotes, reticulate, rmarkdown, rstudioapi, shiny, testthat,
    uuid, waldo, yaml, webfakes
Encoding: UTF-8
RoxygenNote: 7.2.3
Roxygen: list(markdown = TRUE)
VignetteBuilder: knitr
Config/Needs/website: tidyverse/tidytemplate
Config/testthat/edition: 3
Config/testthat/parallel: true
Config/testthat/start-first: bioconductor,python,install,restore,snapshot,retrieve,remotes


RemoteType: github
```

Reply:
Got it. It should be safe to remove all empty lines, right? (Because a valid `DESCRIPTION` will only have empty lines at the end).

Reply:
Yeah exactly.